lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -657,7 +687,8 @@ void ExportVerilog::prepareHWModule(Block &block,
          (op.getAttrs().size() == 1 && op.hasAttr("sv.namehint")))) {
       // Lower this operation to a balanced binary tree of the same operation.
       SmallVector<Operation *> newOps;
-      auto result = lowerFullyAssociativeOp(op, op.getOperands(), newOps);
+      auto result = lowerFullyAssociativeOp(op, op.getOperands(), newOps,
+                                            operandMap, options);
       op.getResult(0).replaceAllUsesWith(result);
       op.erase();
```

Comment:
It would be necessary to erase the original value from `operandMap` before actually deleting the op to avoid UAF. `rewriteAddWithNegativeConstant` function would also erase operations.

Reply:
Great catch. I don't think this should actually happen, but there may be issues if `operandMap` gets resized at some point.